Binary_search.rb

```diff
@@ -0,0 +1,20 @@
+def binary_search(array, target)

```

Comment:
if they ask you to do this in a whiteboard, can you do this without looking at anything?

Reply:
Before i started coding this, i solved this using paper and pen for better understanding. Honestly reading the algorithm or coding it didn't given me much understanding of how it works. so i had to use paper and pen and  solved which actually helped me understand how this algorithm actually works. I'm using this way of understanding on sorting algorithms also